sql/telemetry_derived/ssl_ratios_v1.sql

```diff
@@ -0,0 +1,31 @@
+CREATE TEMP FUNCTION udf_get_key(map ANY TYPE, k ANY TYPE) AS (
+ (
+   SELECT key_value.value
+   FROM UNNEST(map) AS key_value
+   WHERE key_value.key = k
+   LIMIT 1
+ )
+);
+--
+SELECT
+  submission_date,
+  os,
+  country,
+  SUM(udf_get_key(histogram_parent_http_pageload_is_ssl, 0)) AS non_ssl_loads,
+  SUM(udf_get_key(histogram_parent_http_pageload_is_ssl, 1)) AS ssl_loads,
+  -- ratio of pings that have the probe
+  COUNTIF(ARRAY_LENGTH(histogram_parent_http_pageload_is_ssl) > 0) / COUNT(*) AS reporting_ratio
```

Comment:
For some reason the `COUNTIF` here is necessary in tests, since `COUNT` returns the count of all rows.

Reply:
ah, yes. `COUNT` counts `IS NOT NULL` values, and you're right we want to count `> 0` values here.

Reply:
wait...I would expect we want to count `IS NOT NULL` values for `reporting_ratio`

Reply:
It's pretty weird. Running this in the BQ console, `COUNT()` works fine, but from the tests it doesn't skip rows with no values in the array. This field isn't NULLABLE, it's REPEATED, so this `COUNTIF()` is checking if there are any values in the array.

Reply:
ah, most excellent. thanks for the explanation!